src/Flysystem/FlysystemAssetStore.php

```diff
@@ -785,9 +798,37 @@ protected function parseFileID($fileID)
 
         $filename = $matches['folder'] . $matches['basename'] . $matches['extension'];
         $variant = isset($matches['variant']) ? $matches['variant'] : null;
+        $hash = isset($matches['hash']) ? $matches['hash'] : null;
         return [
             'Filename' => $filename,
             'Variant' => $variant,
+            'Hash' => $hash
+        ];
+    }
+
+    /**
+     * Try to parse a file ID using the old SilverStripe 3 format legacy or the SS4 legacy filename format.
+     *
+     * @param string $fileID
+     * @return array
+     */
+    private function parseLegacyFileID($fileID)
```

Comment:
Originally, I had this as part of `parseFileID` with an extra `$forceLegacy` argument. Decided to make it its own method to avoid potentially breaking a class that would extend `FlysystemAssetStore` and override `parseFileID`. Also the addition of the SS3 file pattern made this use case sufficiently different to warrant a separate method.

I made the method private because, I presume I'll be potentially rewriting this in the SS44 implementation and I don't want to introduce new API to support in a patch release.